scripts/generate-config-data.js

```diff
@@ -0,0 +1,245 @@
+      'opposite'
+    ],
+    [
+      /^\|c: &Config\| c.([a-zA-Z_]+).as_ref\(\)\?\.([a-zA-Z_]+)(?:\.clone\(\)|\.as_ref\(\))?\.map\(\|vec\| vec.join\(","\)\)?$/,
+      'array_of'
+    ],
+    [
+      /^\|_\| None$/,
+      'none'
+    ]
+  ];
+
+  for (const [regex, configIs] of regexToConfigIs) {
+    const parts = regex.exec(configCallback);
+    if (parts !== null) {
+      const [, configSection, configProp] = parts;
+      return {configIs, configSection, configProp};
+    }
+  }
+
+  console.warn('Warning: Failed to recognize callback', configCallback);
+  return {configIs: 'unknown'};
+}
+
+function makeCliConfigTree (parsedCliOptions) {
+  const configCliOptions = parsedCliOptions.filter(({configIs}) =>
+        configIs !== 'none' && configIs !== 'unknown'
+    );
+
+  const tree = {};
+  configCliOptions.forEach(({configSection, configProp, ...rest}) => {
+    tree[configSection] = tree[configSection] || {};
+    tree[configSection][configProp] = {...rest};
+  });
+  return tree;
+}
+
+function getStructFields (name, source) {
+  const subsource = source.match(new RegExp(`^struct ${name} {[^]+?}`, 'm'))[0];
+
+  const fields = subsource.split('\n').slice(1, -1);
+
+  const parsedFields = fields.map(line => {
+    const matches = line.match(/^\s*(\w+): Option<([\w<>]+)>,$/);
+    const [, name, type] = matches;
+    return {name, type};
+  });
+
+  return parsedFields;
+}
+
+function hydrateConfigWithCli (config, cliConfig) {
+  const data = {};
+
+  for (const section of config) {
+    data[section.name] = {};
+    data[section.name].section = section.name_friendly;
+
+    for (const prop of section.props) {
+      data[section.name][prop.name] = {};
+      data[section.name][prop.name].name = prop.name.replace(/_/g, ' ').replace(/^./, a => a.toUpperCase());
+      data[section.name][prop.name].type = prop.type;
+
+      // If the config prop is linked to a CLI command
+      if (section.name in cliConfig &&
+          prop.name in cliConfig[section.name]) {
+        const cliConfigProp = cliConfig[section.name][prop.name];
+        const dataProp = data[section.name][prop.name];
+
+        if (cliConfigProp.configIs === 'same') {
+          dataProp.description = cliConfigProp.help;
+          dataProp.default = cliConfigProp.defaultValue;
+        } else if (cliConfigProp.configIs === 'array_of') {
+          dataProp.description = cliConfigProp.help;
+          if (cliConfigProp.defaultValue !== null) {
+            dataProp.default = cliConfigProp.defaultValue.split(',');
+          }
+        } else if (cliConfigProp.configIs === 'opposite') {
+          dataProp.default = !cliConfigProp.defaultValue;
+        } else {
+          console.error("Error: Unreachable: configIs can be 'same', 'array_of', 'opposite', 'none', 'unknown'; is none of the first three; last two are pruned in makeCliConfigTree; qed.");
```

Comment:
Would turn it into hard error (`throw new Error()`) and display `cliConfigProp.configIs` in the error as well.